add a way to post process append 3p files to our js binaries

erwinmombay opened the pull request:
this fixes amp-viz-vega and amp-date-picker which rely on 3p code

erwinmombay commented:
pushed out some additional code @cvializ seems like everything is working. can you PTAL? thanks!

erwinmombay commented on build-system/single-pass.js:
```diff
@@ -521,6 +526,44 @@ function wrapMainBinaries() {
+ * source maps.
+ */
+function postProcessConcat() {
+  const extensions = extensionBundles.filter(
```
@choumx do you mind reviewing this part since you did a bit of the same work around in the multi-pass code.

choumx replied:
Is this meant to subsume `appendToCompiledFile()` in gulpfile.js? I don't see that being removed anywhere.

choumx replied:
In that case, can we call `appendToCompiledFile()` here? We should at least de-dupe.

erwinmombay commented:
@cvializ can you confirm the warnings below are OK?

<img width="848" alt="screen shot 2018-10-30 at 10 18 01 am" src="https://user-images.githubusercontent.com/354746/47737012-252ac880-dc2d-11e8-9fd1-62a988842fde.png">

erwinmombay commented:
actually that first warning is definitely a bug. will fix

erwinmombay commented on examples/date-picker.amp.html:
```diff
@@ -47,7 +47,7 @@
   <script async custom-element="amp-lightbox" src="https://cdn.ampproject.org/v0/amp-lightbox-0.1.js"></script>
   <script async custom-element="amp-date-picker" src="https://cdn.ampproject.org/v0/amp-date-picker-0.1.js"></script>
-  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-latest.js"></script>
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.1.js"></script>
```
@cvializ was there a reason why we used `-latest`?

cvializ replied:
Nope, I think it may have been left in during a local test. I think it would need to be `0.2` in a real amp document

cvializ commented:
The warnings are OK. To check if a date is an RRULE or an ISO8601 date the code uses moment(datestring).isValid() to check if the input is a valid ISO8601 date, and that warning is produced from that. The requestAnimationFrame handler is due to performance, which has been documented

erwinmombay commented:
i don't see that same warning in multi-pass right now so its probably a missed conversion to JsonObject/dict. still searching

erwinmombay commented:
@cvializ i stand corrected. switched to cdn mode. that warning is also there. my bad

erwinmombay commented:
@cvializ i rebased and added your min
<img width="1028" alt="screen shot 2018-10-31 at 12 17 36 am" src="https://user-images.githubusercontent.com/354746/47771994-804ad280-dca2-11e8-966f-ad2c5b08d6f6.png">
 max changes and now getting these errors using cdn mode

erwinmombay commented:
@cvializ thanks for explaining
